mempool/clist_mempool_test.go

```diff
@@ -292,14 +292,33 @@ func TestMempoolUpdate(t *testing.T) {
 	}
 }
 
+func TestMempoolLanesValidation(t *testing.T) {
+	appInfo := abci.InfoResponse{}
+
+	_, err := FetchLanesInfo(appInfo.LanePriorities, types.Lane(appInfo.DefaultLanePriority))
+	require.NoError(t, err)
+
+	_, err = FetchLanesInfo(appInfo.LanePriorities, types.Lane(1))
```

Comment:
Should it not be invalid if `lanes` is empty? Because at this point, `appInfo.LanePriorities` has no items. I thought that in that case only a default priority of 0 was acceptable, am I wrong?

Reply:
The value 0 for lanes (including the default lane) is reserved for when we do CheckTx and there are no lanes, or when the tx is invalid, so the app returns 0 on the lane field of the CheckTx response. That's why the default lane is 1.

Reply:
I see, I misunderstood that from the ADR. But still if the list is empty and the default priority is anything but 0 it should still fail, just maybe with the error that the lane is not in the list?

Reply:
Ok, probably that's better.

Reply:
My opinion is that someone calling `FetchLanesInfo` with these parameters (empty list and non-zero default) doesn't really know what they are doing. So I'd rather warn them with an error (it might be a silly mistake that is better if we point it out), even if we could theoretically deal with that _weird_ input.

Reply:
I agree to make it fail with an empty list and non-zero default lane.

Reply:
This is solved then with this test, resolving the conversation.